### PR TITLE
fix: 1038 update the agency selection input

### LINKED ISSUE #1038: update the agency selection input
Add the functionality for a user to add their Acronym and pull in the correct client ID information. 

1) update FE to new component design, show success and error state
2) Complete testing across all screens
3) update client selection tool to add information when an option is selected

### PR DESCRIPTION
**Pull request by mgaseta**
# Description
Closes #1038

#### New
- Added new validation to the agency selection field
- Added new error banner
- Added new error messages
- Added new loading display inside the input

#### Changed
- Changed the agency selection method from a dropdown to an input
- Changed helper texts

#### Removed
- Nothing :)

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-46-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1096-backend.apps.silver.devops.gov.bc.ca/actuator/health)
- [Oracle-API](https://nr-spar-1096-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Review by ngunner15 — APPROVED** [review]
LGTM! 👍🏼

**Review by RMCampos — APPROVED** [review]
Great job! No changes other then the ones Craig is requesting 🎉

**Review by craigyu — CHANGES_REQUESTED** [review]
Good work! ✅
As discussed earlier, a few things needs to be updated:
- leading 0 on single digit location code
- cy.wait() usage
- Green and red circle appearing at the same time

File: frontend/cypress/e2e/smoke-test/create-a-class-seedlot.cy.ts [hunk 1]
```diff
@@ -255,10 +257,11 @@ describe('Create A-Class Seedlot', () => {
...
     // Enter the applicant agency name
-    cy.get('#applicant-info-combobox')
-      .click();
-    cy.contains(`.${prefix}--list-box__menu-item__option`, regData.agencyName)
-      .click();
+    cy.get('#applicant-info-input')
+      .clear()
+      .type(regData.agencyAcronym, { delay: TYPE_DELAY })
+      .blur();
+    cy.wait(FIVE_SECONDS);
```
[the remark this answers is not in the document]
Comment (mgaseta) [reply frontend/cypress/e2e/smoke-test/create-a-class-seedlot.cy.ts]: Fixed here: https://github.com/bcgov/nr-spar/pull/1096/commits/258028721957ae15a14cb5b6e9a1278bb48242b5

**Review by craigyu — APPROVED** [review]
Good job! Surprisingly there is no conflict with Nisarg changes ✌️

**Pull request merged**